docarray/array/stacked/array_stacked.py

```diff
@@ -98,6 +98,20 @@ def from_iterable_document(
             self._docs, tensor_type=self.tensor_type
         )
 
+        self._add_ref_to_docs()
+
+    def _add_ref_to_docs(self) -> None:
+        """Add a ref to self to the document"""
+        from docarray.base_document.document import _DaRef
+
+        for i, doc in enumerate(self._docs):
+            if doc._da_ref is not None and doc._da_ref.ref is not self:
+                raise ValueError(
+                    f'{doc} already belong to {doc._da_ref} and cannot be stack again'
```

Comment:
```suggestion
                    f'{doc} already belongs to {doc._da_ref} and cannot be stacked again'
```
Maybe I would say something like `doc is a view of [...] and cannot be stacked again`